### ISSUE TITLE
Erroneous "Unindent does not match any outer indentation level"

### ISSUE THREAD

**Issue by joseppinilla** [nteract/hydrogen#1224]
**Description:**

Using the new version of Atom, if I want to run code that uses indentation (definition of a method, for loop, conditionals) the kernel is executing all the code or failing with the message "Unindent does not match any outer indentation level".

**Steps to Reproduce:**

1. Start Python 3/2.7 Kernel
2. Place cursor on the definition of a method, or something that uses indentation:

```
def my_method(): #Cursor here
    pass

if True: #Or here
    pass

for i in [1,2]: #Or here
    pass
```

3. Hit shift-enter to execute line.

**Versions:**

1.24.0 x64
Windows 10

Using version 1.23.3 x64 in the same system works fine.
(I went to the atom folder to run the version 1.23.3 that is stored)

**Logs:**

Please post any error logs and the output of the developer tools as described in our [Debugging Guide](https://nteract.gitbooks.io/hydrogen/docs/Troubleshooting.html).

**Comment by nikitakit** [comment]
Could you post an example of actual code that has this issue? (If you don't know how GitHub comment formatting works, you can paste some code into the comments, select it, and press the "Insert Code" button to have it be formatted as a code block)

Running indented code blocks should work fine most of the time, so a specific code block would help narrow down the issue.

**Comment by nikitakit** [comment]
Huh, this does look related to an Atom update. Though I'm not on the latest version of atom myself, at the moment.

BTW even on Atom, 1.23.2, running an indented code block doesn't work if there's a comment after the `:`. It only works if the `:` is the last symbol on that line.

**Comment by nikitakit** [comment]
Does this issue still occur using the latest hydrogen update (2.2.3)?

**Comment by sunt05** [comment]
This issue has been resolved in the recent update of Hydrogen (2.2.3).
